packages/arb-rpc-node/batcher/sequencerBatcher.go

```diff
@@ -739,6 +751,15 @@ func (b *SequencerBatcher) publishBatch(ctx context.Context, dontPublishBlockNum
 		return false, err
 	}
 
+	if publishingAllBatchItems {
+		// Reset the pending gas estimate to gasCostBase,
+		// plus whatever was sequenced while we were building this batch.
+		atomic.AddInt64(&b.pendingBatchGasEstimateAtomic, int64(gasCostBase)-origEstimate)
+	} else {
+		// Since we didn't publish everything, only subtract gas for what we did publish.
+		atomic.AddInt64(&b.pendingBatchGasEstimateAtomic, -int64(estimatedGasCost))
```

Comment:
In this case, doesn't the base gas cost of the next batch still need to be incorporated into the total?

Reply:
Ah, good call